stackslib/src/net/api/postblock_proposal.rs

```diff
@@ -541,8 +557,67 @@ impl NakamotoBlockProposal {
             builder.load_tenure_info(chainstate, &burn_dbconn, tenure_cause)?;
         let mut tenure_tx = builder.tenure_begin(&burn_dbconn, &mut miner_tenure_info)?;
 
+        let mut replay_txs_maybe: Option<VecDeque<StacksTransaction>> =
+            self.replay_txs.clone().map(|txs| txs.into());
+
         for (i, tx) in self.block.txs.iter().enumerate() {
             let tx_len = tx.tx_len();
+
+            // Check if tx is in the replay set, if required
+            if let Some(ref mut replay_txs) = replay_txs_maybe {
+                loop {
+                    let Some(replay_tx) = replay_txs.pop_front() else {
+                        return Err(BlockValidateRejectReason {
```

Comment:
To clarify, the idea is that it is not allowed to have a block that includes all of the replay transactions, followed by additional transactions?

Reply:
If that is the case, can you add a comment here, because it is a little confusing.

Reply:
yep exactly. Added a comment in [bb7e57b](https://github.com/stacks-network/stacks-core/pull/5994/commits/bb7e57b4d6a580893a890400a640a9f35c0cc619)